common/parallel/TreeCrawler.go

```diff
@@ -98,18 +98,13 @@ func (c *crawler) runWorkersToCompletion(ctx context.Context) {
 func (c *crawler) workerLoop(ctx context.Context, wg *sync.WaitGroup, workerIndex int) {
 	defer wg.Done()
 
-	var err error
 	mayHaveMore := true
 	for mayHaveMore && ctx.Err() == nil {
-		mayHaveMore, err = c.processOneDirectory(ctx, workerIndex)
-		if err != nil {
-			c.output <- CrawlResult{err: err}
-			// output the error, but we don't necessarily stop the enumeration (e.g. it might be one unreadable dir)
-		}
```

Comment:
processOneDirectory() is not returning any other errors except bodyErr (errors from workerBody() which is enumerateOneFileSystemDirectory()) I thought its better to deal with those errors in enumerateOneFileSystemDirectory() instead of having to deal with them at two places.

Reply:
For local filesystem enumeration, where workerBody() is enumerateOneFileSystemDirectory() that doesn't make a difference, but processOneDirectory() is a general function used for single-directory-enumeration, f.e. we have blobTraverser.parallelList() for flat blobs and fileTraverser.traverse() for azure files both of which also call parallel.Crawl() which will end up calling processOneDirectory() with a different workerBody(). Those return err hoping that it will be passed to the caller. Now with your change removing the error return from processOneDirectory() that will no longer work.
What I'm suggesting is a way in which we can support those workerBody which do not care about enqueuing a high-fidelity FileSystemEntry object (with filepath and fileinfo) but just want to return a generic error, and also those which want to enqueue FileSystemEntry  with all details.
Let me know if you want to discus..

Reply:
I agree this needs to remain same for other enumerators. enumerateOneFileSystemDirectory() returns nil and that ensures there wont be duplicates in queue. Feels kinda upside down but i think this is the right approach